c/tskit/tables.c

```diff
@@ -468,6 +468,48 @@ check_offsets(
     return ret;
 }
 
+static int
+calculate_max_rows(tsk_size_t num_rows, tsk_size_t max_rows,
+    tsk_size_t max_rows_increment, tsk_size_t additional_rows,
+    tsk_size_t *ret_new_max_rows)
+{
+    tsk_size_t new_max_rows;
+    int ret = 0;
+
+    if (check_table_overflow(num_rows, additional_rows)) {
+        ret = TSK_ERR_TABLE_OVERFLOW;
+        goto out;
+    }
+
+    if (num_rows + additional_rows <= max_rows) {
+        new_max_rows = max_rows;
+    } else {
+        if (max_rows_increment == 0) {
+            /* Doubling by default */
+            new_max_rows = TSK_MIN(max_rows * 2, TSK_MAX_ID + (tsk_size_t) 1);
+            /* Add some constraints to prevent very small or very large allocations */
+            if (new_max_rows < 1024) {
+                new_max_rows = 1024;
+            }
+            if (new_max_rows - max_rows > 2097152) {
+                new_max_rows = max_rows + 2097152;
+            }
+            new_max_rows = TSK_MAX(new_max_rows, num_rows + additional_rows);
+        } else {
+            /* Use user increment value */
```

Comment:
Looks like this overflow branch isn't tested?

Reply:
Fixed with an extra test